src/openms/include/OpenMS/FILTERING/CALIBRATION/PrecursorCorrection.h

```diff
@@ -51,58 +51,83 @@
 
 namespace OpenMS
 {
-
+  /**
+  @brief A class with functions for precursor correction
+  */
 class OPENMS_DLLAPI PrecursorCorrection
 {
     public:
     static const std::string csv_header;
 
-    static void getPrecursors(
-        const MSExperiment & exp, 
-        std::vector<Precursor> & precursors, 
-        std::vector<double> & precursors_rt, 
-        std::vector<Size> precursor_scan_index);
-    
-    static void writeHist(const String& out_csv, const std::vector<double> & deltaMZs, const std::vector<double> & mzs, const std::vector<double> & rts);
-
-    static std::set<Size> correctToNearestMS1Peak(
-      MSExperiment & exp, 
-      double mz_tolerance, 
-      bool ppm, 
-      std::vector<double> & deltaMZs, 
-      std::vector<double> & mzs, 
-      std::vector<double> & rts);
-
-    //Selection of the peak with the highest intensity as corrected precursor mass in a given mass range (e.g. precursor mass +/- 0.2 Da)
-    static std::set<Size> correctToHighestIntensityMS1Peak(
-      MSExperiment & exp, 
-      double mz_tolerance, 
-      std::vector<double> & deltaMZs, 
-      std::vector<double> & mzs, 
-      std::vector<double> & rts);
-
-    // Wrong assignment of the mono-isotopic mass for precursors are assumed:
-    // - if precursor_mz matches the mz of a non-monoisotopic feature mass trace
-    // - and in the case that believe_charge is true: if feature_charge matches the precursor_charge
-    // In the case of wrong mono-isotopic assignment several options for correction are available:
-    // keep_original will create a copy of the precursor and tandem spectrum for the new mono-isotopic mass trace and retain the original one
-    // all_matching_features does this not for only the closest feature but all features in a question
-    static std::set<Size> correctToNearestFeature(
-      const FeatureMap& features, 
-      MSExperiment & exp, 
-      double rt_tolerance_s = 0.0, 
-      double mz_tolerance = 0.0, 
-      bool ppm = true, 
-      bool believe_charge = false, 
-      bool keep_original = false, 
-      bool all_matching_features = false, 
-      int max_trace = 2,
-      int debug_level = 0);
+    // TODO: add comments
+    /**
+    @brief Get precursor information
```

Comment:
maybe rename this method? might also need some comments if/how different MS levels are supported etc.